apps/dcos_overlay/test/dcos_overlay_SUITE.erl

```diff
@@ -179,126 +77,53 @@ meck_httpc_request(get, {_, Headers}, _, [{sync, false}]) ->
-  Cover = stop_cover_enode(ENode),
-  spawn(ENode, init, stop, []),
-  case wait_for_node_dead(ENode, 60) of
-    {ok, ENode} ->
-      maybe_signal_cover_master(ENode, Cover),
-      {ok, ENode};
-    Error ->
-      Error
-  end.
-
-stop_cover_enode(ENode) ->
-  case test_server:is_cover() of
-    true ->
-      Main = cover:get_main_node(),
-      rpc:call(Main, cover, flush, [ENode]),
-      {true, Main};
-    false ->
-      {false, undefined}
-  end.
-
-%% To avoid that cover is started again if a node
-%% with the same name is started later.
-maybe_signal_cover_master(ENode, {true, MainCoverNode}) ->
-  rpc:call(MainCoverNode, cover, stop, [ENode]);
-maybe_signal_cover_master(_, {false, _}) ->
-  ok.
-
-% wait until timeout N seconds until node is disconnected
-% relies on disterl to tell us if a node has died
-% Maybe we should net_adm:ping?
-wait_for_node_dead(Node, 0) ->
-  {error, stop_timeout, Node};
-wait_for_node_dead(Node, N) ->
-  timer:sleep(1000),
-  case lists:member(Node, nodes()) of
-    true ->
-      wait_for_node_dead(Node, N - 1);
-    false ->
-      {ok, Node}
-  end.
-
-stop_nodes(Nodes) ->
-  StoppedResult = [do_stop(Node) || Node <- Nodes],
-  ct:pal("Stopped result: ~p", [StoppedResult]),
-  [maybe_kill(Node) || Node <- Nodes].
-
-
 dcos_overlay_test(Config) ->
-  AllNodes = ?config(agents, Config) ++ ?config(masters, Config),
-  rpc:multicall(AllNodes, application, ensure_all_started, [dcos_overlay]),
-  timer:sleep(60000),  %% let the system configure
-  ok = check_files(?NUM_OF_TRIES, Config),
-  ok = check_data(?NUM_OF_TRIES, Config).
-
-check_files(N, Config) when N > 0 ->
-  NumNodes = ?AGENT_COUNT + ?MASTER_COUNT + 1, % agents, masters and runner
-  {ok, Filenames} = file:list_dir(?OUTPUTDIR(?config(priv_dir, Config))),
-  io:format("Files: ~p", [Filenames]),
-  case length(Filenames) of
-      NumNodes ->
-        ok;
-      _ ->
-        timer:sleep(5000), % give time for file generation
-        check_files(N-1, Config)
-  end;
-check_files(_, _) ->
-  not_ok.
-
-check_data(N, Config) when N > 0 ->
-   case verify_data(Config) of
-       [] ->
-         ok;
-       _ ->
-         timer:sleep(5000),
-         check_data(N-1, Config)
-     end;
-check_data(_, _) ->
-  not_ok.
-
-verify_data(Config) ->
-  [Master|_] = ?config(masters, Config),
-  Filename = filename:join(?OUTPUTDIR(?config(priv_dir, Config)), Master),
-  {ok, Actual} = file:consult(Filename),
-  Expected = expected_data(Master),
-  io:format("Actual ~p\n", [Actual]),
-  io:format("Expected ~p\n", [Expected]),
-  Result = ordsets:subtract(ordsets:from_list(Expected), ordsets:from_list(Actual)),
-  io:format("Result ~p\n", [Result]),
-  Result.
-
-expected_data(Master) ->
-  NumNodes = ?AGENT_COUNT + ?MASTER_COUNT,
-  MasterNode = parse_node(list_to_binary(atom_to_list(Master))),
-  [ get_node_data(N) || N <- lists:seq(1, NumNodes), N =/= MasterNode].
+    Nodes = ?config(nodes, Config),
+    {_, []} = rpc:multicall(Nodes, meck, new,
```

Comment:
@GoelDeepak could you please try to move all `meck:new/2` and `meck:expect/2` calls to new `meck_setup/1` function and call that function on all `Nodes` using `erlang:apply/2`?

```
rpc:multicall(Nodes, erlang, apply, [fun meck_setup/1, []])
```